### DISCUSSION TITLE
My Workflow for giving copilot more context

### DISCUSSION THREAD

**Discussion by busti** [community/community#9844] [Copilot Conversations]
As far as I know copilot does not work across multiple files in a project.  
So to get better recommendations, I have started to simply copy lots of code from other files into the one I am working on, which significantly improves the recommendations. 

Is anyone else doing the same?

**Comment by MrMage** [comment]
I am doing similar things, like temporarily copying my model definition into a view in order to have Copilot better complete rendering for additional model fields. It seems to also take into account other open files from the workspace (see https://github.com/github/feedback/discussions/7573#discussioncomment-1930673), but that can be hit-or-miss. Can't wait till Copilot can simply learn from the rest of my repository.

For porting a codebase from Swift to PHP, I also copied in snippets of "successful" conversions to give Copilot more context for how it should convert the code.

**Reply by busti** [reply]
I also find myself writing example code that is close to what I want copilot to generate to give it some hints, which I end up deleting later.
